pykdtree/kdtree.pyx

```diff
@@ -94,6 +94,8 @@ cdef class KDTree:
         # Check arguments
         if leafsize < 1:
             raise ValueError('leafsize must be greater than zero')
+        if data_pts.size == 0:
+            raise ValueError('data_pts should be non-empty')
```

Comment:
The docstring says `data_pts` should be of shape `(n, dims)`. What is a reasonable minimum size for this array? 2 elements in a 1D case? I'm wondering if this could be a `< 1` check *or* it could maybe be a check on the `.ndim` being 2 and then check of the first dimension being greater than 1 (number of points). Thoughts?

Reply:
Yep, I like it. However I'd like to leave `.size == 0` as the second check just because it's the same but a bit more concise and pretty than `.shape[1] > 0` as for me.
Btw why 2 elements? In 1D case it could be something like `np.array([[1.0]])`

Reply:
I guess I was thinking more logically than just functionally. The code might work with one point, but is the user actually querying over just 1 point? But yeah, let's not restrict the possible uses.